MoltenVK/MoltenVK/GPUObjects/MVKDevice.h

```diff
@@ -907,6 +931,9 @@ class MVKDevice : public MVKDispatchableVulkanAPIObject {
 	MVKSmallVector<MVKSmallVector<MVKQueue*, kMVKQueueCountPerQueueFamily>, kMVKQueueFamilyCount> _queuesByQueueFamilyIndex;
 	MVKSmallVector<MVKResource*, 256> _resources;
 	MVKSmallVector<MVKBuffer*, 8> _gpuAddressableBuffers;
+    std::unordered_map<MVKBufferAddressRange, MVKBuffer*, MVKHash_uint64_t_pair> _gpuBufferAddressMap;
```

Comment:
The perfectionist in me wants to use a data structure specifically designed for mapping ranges to values. There's gotta be such a structure out there somewhere...

Reply:
Something like a IntervalTree https://en.m.wikipedia.org/wiki/Interval_tree ? Java can use something like RangeMap https://www.baeldung.com/guava-rangemap . In cpp i believe boost has something.

Reply:
Boost is probaly to heavy for use in MoltenVK, unless it's standalone. I found an interval tree implementation on github [here](https://github.com/5cript/interval-tree). This could be something to consider for the future, but I'm not sure if we have to implement it immediately.